packages/javascript/bh-shared-ui/src/views/Explore/ContextMenu/AssetGroupMenuItemPrivilegeZonesEnabled.test.tsx

```diff
@@ -35,178 +92,118 @@ const getEntityInfoTestProps = () => ({
+            })
+        );
+    }),
+    rest.post('/api/v2/asset-group-tags/:tagId/selectors', (req, res, ctx) => {
+        return res(ctx.json({}));
+    })
+);
 
-const AssetGroupMenuItemWithDialog = (props: any) => {
-    const [dialogOpen, setDialogOpen] = useState(false);
-
-    return (
-        <Dialog open={dialogOpen}>
-            <AssetGroupMenuItem
-                {...props}
-                onShowConfirmation={() => setDialogOpen(true)}
-                onCancelConfirmation={() => setDialogOpen(false)}
-            />
-        </Dialog>
-    );
-};
+const ROUTE_WITH_SELECTED_ITEM_PARAM = `?selectedItem=${getEntityInfoTestProps().entityinfo.selectedNode.id}&searchType=node&primarySearch=${getEntityInfoTestProps().entityinfo.selectedNode.id}`;
 
 describe('AssetGroupMenuItem', () => {
-    describe('adding to an asset group', () => {
-        const server = setupServer(
-            rest.get('/api/v2/graph-search', (req, res, ctx) => {
-                return res(ctx.json({}));
-            }),
-            rest.post('/api/v2/graphs/cypher', (req, res, ctx) => {
-                return res(ctx.json({}));
-            }),
-            rest.post('/api/v2/asset-group-tags/:tagId/selectors', (req, res, ctx) => {
-                return res(ctx.json({}));
-            })
-        );
+    beforeAll(() => server.listen());
 
-        beforeAll(() => server.listen());
+    afterEach(() => {
+        server.resetHandlers();
+        vi.restoreAllMocks();
+    });
 
-        afterEach(() => {
-            server.resetHandlers();
-        });
+    afterAll(() => server.close());
 
-        afterAll(() => server.close());
-
-        it('handles adding to tier zero asset group', async () => {
-            const testOnAddNode = vi.fn();
-            const testRemoveNodePath = '/meow';
-
-            render(
-                <AssetGroupMenuItemWithDialog
-                    assetGroupId={tierZeroAssetGroup.id}
-                    assetGroupName={tierZeroAssetGroup.name}
-                    isCurrentMember={false}
-                    showConfirmationOnAdd={true}
-                    onAddNode={testOnAddNode}
-                    removeNodePath={testRemoveNodePath}
-                />,
-                {
-                    route: ROUTE_WITH_SELECTED_ITEM_PARAM,
-                }
-            );
+    describe('adding to an asset group', () => {
+        it('adds node to tier zero asset group', async () => {
+            const mutateSpy = vi.spyOn(apiClient, 'createAssetGroupTagSelector');
+
+            render(<AssetGroupMenuItem assetGroupType='tierZero' showConfirmationOnAdd={true} />, {
+                route: ROUTE_WITH_SELECTED_ITEM_PARAM,
+            });
 
             const user = userEvent.setup();
 
-            const addToHighValueButton = screen.getByRole('menuitem', { name: /add to high value/i });
+            const addToHighValueButton = await screen.findByRole('menuitem', { name: /Add to Tier Zero/i });
```

Comment:
`await` + `findByRole` ensures that all the TanStack queries are resolved before running assertions.